components/script/dom/htmlmediaelement.rs

```diff
@@ -1217,6 +1250,37 @@ impl HTMLMediaElement {
                 //    an unsupported format, or can otherwise not be rendered at all"
                 if self.ready_state.get() < ReadyState::HaveMetadata {
                     self.queue_dedicated_media_source_failure_steps();
+                } else {
+                    // https://html.spec.whatwg.org/multipage/#reaches-the-end
+                    match self.direction_of_playback() {
+                        PlaybackDirection::Forwards => {
+                            // Step 1.
+                            if self.Loop() {
+                                self.seek(
+                                    self.earliest_possible_position(),
+                                    /* approximate_for_speed*/ false,
+                                );
+                            } else {
+                                // Step 2.
+                                // The **ended playback** condition is implemented inside of
+                                // the HTMLMediaElementMethods::Ended method
+
+                                // Step 3.
+                                self.queue_reaches_the_end_steps();
```

Comment:
Since this is only going to be used here, could we inline the implementation of `queue_reaches_the_end_steps` here?

Reply:
Yes, the reason I put it there is because I looked for examples of using the queue and tried to follow the same pattern. But it this case I'm splitting the steps between two methods, which probably hurts more than it helps..

Thanks for the feedback!